### PR TITLE
Cart n receipt

### LINKED ISSUE #14: Purchasing souvenirs on a trip
### Description:
As a Baseball Obsessed fan with deep pockets, I'd like the ability to purchase multiple souvenirs at each location I visit.

### Assumptions:
* The use if able to take a trip
* Classes are made and data is being loaded into he program

### Estimate:
* 5

### Tasks:
- [ ] Create a cart to place items into
- [ ] The receipt must show the number of items bought
- [ ] Have to be able to add or remove items from that cart 
- [ ] Must show a "receipt" after a purchase at each stadium
- [ ] Must also have a grand total receipt that shows all the purchases made at each stadium on the trip with the total amount of money spent at all stadiums

### Tests:
* Test to see the the user is able to add and delete items at each Stadium
* Test to see that when completing a transaction the correct number of items bought displays on the immediate receipt at each restaurant
*  Test to see that when completing a transaction the correct number of items bought displays once the trip is concluded on the grand total receipt for the trip

### Definition of Done:
* The user is able to add items to cart and the user can edit their cart at any point by adjusting the qty, deleting and can add more items to the cart if they wish. 
* The user checks out and the correct item qty show on the immediate receipt and total receipt

### LINKED ISSUE #18: Custom trip from Angel's Stadium and stadium of choice
### Description:
As a baseball fan vacationer, I want to visit all of the stadiums I desire starting from both my choice of stadium(requirement 2, 4) or starting from Angels Stadium in the most efficient order (requirement 1) as well as the see the total distance of the trip so that I can plan my trip of dreams

### Assumptions:
* The main window is initialized
* The stadiums and teams have been loaded into the program
* The weighted graph is implemented
* The algorithms for the trip are implemented

### Estimate:
* 18

### Tasks:
* Create a view for planning trips
* Prompt the user for the trip they wish to plan
* Prompt the user for the stadiums they wish to visit
* Calculate the shortest path based off the prewritten algorithm that corresponds to the type of trip
* Allow the user to purchase souvenirs
* Calculate the total distance of the trip based off of the shortest path the prewritten algorithm calculates
* Display the stadiums, the team names, the souvenirs purchased, the grand total of the amount spent, and the total distance of the trip

### Tests:
* Verify that the stadiums selected are put into the planned trip
* Verify that stadiums that are not selected are not put into the planned trip
* Verify that the stadiums are sorted in the most efficient way for the trip
* Verify that the distances are calculated correctly for the trip
* Verify that all of the totals are displayed correctly

### Definition of Done:
This story is considered done when the user can plan a trip from Angel's stadium or a stadium of their choice and visit all of the stadiums that they desire in the most efficient manner corresponding to the algorithm of the trip and all of the totals (distance and money spent) are displayed correctly

### LINKED ISSUE #19: Visit all teams from Comerica Park (Detroit Tigers) choosing shortest distances along the way
### Description:
As a Tigers fan and literal cross country runner, I would like to visit all of the teams in the order of closest stadium to closest stadium and then see the total distance traveled so that I can run to each of these locations only having to do the minimum distances per day of my trip.

### Assumptions:
* There exists stadiums in the weight graph and the graph exists. If not then display that no stadiums are present
* There exists means of traversing the graph and weighted links between nodes
* There also exists a way of displaying total distance traveled 

### Estimate:
* 5

### Tasks:
1. Add a way of requesting input to when the traversal should occur
2. Perform the actual traversal
3. Display the output of the total distance traveled and the path it took

### Tests:
1. Verify that the path the search took is correct
2. Verify that the total distance is correct
3. Verify that the output is correct 

### Definition of Done:
This story is considered done when I can click on something and I get the path taken by this algorithm visiting all teams starting from Comercia Park and see the total distance traveled

### PR DESCRIPTION
**Pull request by Roger-Chavez**
Closes #14
Closes #18
Closes #19

### Key features
* You can make a 4 different types of trips
* buy items at each stadium you visit
* You can add and remove items from your cart
* You can see how many items you have in your cart, updates in real-time
* You can see the total distance traveled
* You can see the grand total spent within your trip
*  You can see a preview of a trip before you take it

### Future Improvements
* Need to edit ui on windows, mac ui settings don't match windows

### Notes
none

### CODE REVIEW

File: app/src/windows/mainwindow.ui [hunk 1]
```diff
@@ -88,7 +130,7 @@ QTreeWidget
      <string notr="true"/>
     </property>
     <property name="currentIndex">
-     <number>0</number>
+     <number>2</number>
```
Comment (ChrisRzech) [inline app/src/windows/mainwindow.ui]: :hmm:

Comment (Roger-Chavez) [reply app/src/windows/mainwindow.ui]: :blaze:

File: app/src/widgets/souvenirlist.cpp [hunk 2]
```diff
@@ -190,9 +190,9 @@ void SouvenirList::removeItem(IDs id)
 
         if(widget != nullptr && id == widget->getIDs())
         {
-            emit itemQtyChanged(id, widget->getIDs().second);
             m_IDQtys.erase(widget->getIDs());
             QListWidget::takeItem(row);
+            emit itemQtyChanged(id, -widget->getIDs().second);
```
Comment (ChrisRzech) [inline app/src/widgets/souvenirlist.cpp]: Wuh oh, remove all the changes in this file

Comment (Roger-Chavez) [reply app/src/widgets/souvenirlist.cpp]: oh my bad, had to test it on this branch to make sure it worked

Comment (Roger-Chavez) [reply app/src/widgets/souvenirlist.cpp]: oh actually no, need to make a separate branch for it, these changes are how it's supposed to be

**Comment by ChrisRzech** [comment]
The closing statement on the desc is wrong, please fix

**Review by yosmo78 — APPROVED** [review]
Looks good, just make sure when you merge that Anthony's graph view is still present and working

**Review by ChrisRzech — APPROVED** [review]
🙏 let's hope

**Pull request merged**